spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/CommitOwnerClient.scala

```diff
@@ -43,7 +43,7 @@ case class Commit(
 case class CommitFailedException(
     private val retryable: Boolean,
     private val conflict: Boolean,
-    private val message: String) extends Exception(message) {
+    private val message: String) extends RuntimeException(message) {
```

Comment:
Why is this changed needed? Isn't that just making it more restrictive?

Reply:
This is needed because the CommitOwnerClient trait does not declare that `commit` can throw `CommitFailedException` in the signature (Scala doesn't let us do that). The Java code won't compile if I try to throw an exception of type Exception if it has not been declared in the function signature. `RuntimeException`s can be thrown though.